addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3152,7 +3155,7 @@ class SnippetsMenu extends Component {
 
                 // Mark any snippet that is not custom as raw as they should
                 // not change during edition.
-                this.snippets.set(snippet.id, snippet.isCustom ? snippet : markRaw(snippet));
+                this.snippets.set(snippet.id, snippet);
```

Comment:
`markRaw(snippet)` caused a significant delay when enabling/disabling them in the menu, because the object was not reactive. It actually needs to be reactive in that case (specifically the `snippet.data.visible` attribute). Doing that causes just 1 more rerender for all snippets (not N renders for N snippets).

Reply:
@robinlej @detrouxdev Not sure to understand what's being discussed here :grimacing: (is this PR ready?)

Reply:
`markRaw` prevents objects from being turned reactive. I thought that non-custom snippets probably would not have any property that would change during edition, so I marked them as "non-reactive" (as opposed to custom snippets since their name and edition state can change). But this meant that if we disabled a snippet after they were initially rendered (which is what we do here), we would not see that change until something else triggered a re-render.

Reply:
Ok, it should indeed be reactive now I guess: although was not it the case before already? The disabled state could change?

> (is this PR ready?)

It's not I guess @robinlej since this comment is not right :wink: 
```javascript
                // Mark any snippet that is not custom as raw as they should
                // not change during edition.
                this.snippets.set(snippet.id, snippet);
```

Reply:
> Ok, it should indeed be reactive now I guess: although was not it the case before already? The disabled state could change?

I don't think it could @qsm-odoo  (I can't find a concrete example where it did). It looks like the change only ever happened in conjunction to a state update, triggering a rerender (including for `snippet.disabled`).